### ISSUE TITLE
AndroidX support

### ISSUE THREAD

**Issue by cckroets** [greenrobot/EventBus#552]
Now that Android Studio 3.2.0 and `androidx` 1.0.0 libraries have been released, is there a plan to migrate to `androidx`? Google has said there won't be any more feature releases under the `android.support` packaging.

**Comment by greenrobot-team** [comment]
Please refer to https://developer.android.com/jetpack/androidx/migrate on how to use AndroidX with libraries that do not yet use the new imports.

Keeping this open to track switching to AndroidX.
-ut

**Comment by themockgraeme** [comment]
I've been having an issue with AndroidX whereby I get a NoClassDefFoundError when the find methods code walks down the inheritance tree.

A quick and dirty fix would be to filter for androidx packages in the moveToSuperclass() method. I've created a pull request for this, and it'd make sense from a performance perspective too.

**Comment by cckroets** [comment]
AndroidX has been out for over a year now. Is there any update on this? This is the last non-AndroidX library in my project's external dependencies.

**Comment by greenrobot-team** [comment]
https://github.com/greenrobot/EventBus/releases/tag/V3.2.0

And the remaining code that uses the legacy Android Support Library APIs should not be used, anyhow (e.g. ErrorDialogManager).

@cckroets I use the latest release in an AndroidX project without issues. What is preventing you from doing so?

**Comment by cckroets** [comment]
It requires enabling Jetifier which adds time to the build. It will analyze and re-write every external dependency to use AndroidX, which can be expensive for large projects.

**Comment by Malachiasz** [comment]
Any plan to support AndroidX? Why the pull request is being ignored https://github.com/greenrobot/EventBus/pull/627 ? 

We want to get rid of Jetifier which slows down builds, so is there a chance that EventBus will get migrated or should we rather remove it from our dependencies?

**Comment by bigcal6977** [comment]
2600:1010:b114:af8d:6c88:3847:a3a6:1e2e

**Comment by jingibus** [comment]
In order to improve our build times (which are a pain point in our codebase), we're currently pushing to remove jetifier from our builds. This change has had two years to bake, and apart from the build speed wins we're excited to simplify our system by eliminating this legacy concern.

We don't directly rely on EventBus, but it is a transitive dependency used by a third party library we use. Unfortunately, jetifier cannot be selectively turned on and off for each individual project: it's either all or nothing. So EventBus is blocking this goal for us.

**Comment by pvegh** [comment]
> And the remaining code that uses the legacy Android Support Library APIs should not be used, anyhow (e.g. ErrorDialogManager).

Then please release 3.3 removing those classed. It's high time!
I want to remove jetifier for better build perf as well. Eventbus is standing in the way.

**Comment by pvegh** [comment]
Anybody there?
A month without answer. 2 years ignoring the problem.
We have now decided to remove EventBus from our project.

**Comment by greenrobot-team** [comment]
Sorry, busy with other things.

Using AndroidX would require making EventBus into an Android library (AAR), have not had time to look into resolving this (e.g. there still should be a plain Java version).

**Comment by Malachiasz** [comment]
But should it really be available as plain Java? I know that it would be nice, but does the library have substantial amount of non-android users?

**Comment by Amrsatrio** [comment]
I do use this libary to share business logic code between my JVM Discord Bot and Android App. So this library should be kept compatible for plain Java in my opinion.

**Comment by andob** [comment]
maybe would be better to just remove all android dependencies, and make this library a pure java library. no need for jetifier or androidx then.

what is the plan for this?
the pull request with androidx conversion is ignored, would a pull request with conversion to pure java be welcomed? :D

**Comment by andob** [comment]
@greenrobot @greenrobot-team

**Comment by andob** [comment]
@pvegh sorry to hear that.

well I do care, I offered a solution, but they didn't respond.. :|

If they don't have time to maintain this library I am even willing to take over the responsability of maintaining.. For me it's not a solution to abandon this library, since I use it extensively.

Anybody there @greenrobot-team ??

**Comment by sarbyn** [comment]
Any update about this issue? Like others developers, this is the last legacy dependencies in our project.
Can someone suggest us a modern alternatives to eventBus for android project?

**Comment by Amrsatrio** [comment]
The recommended solution for that right now is Kotlin Coroutines but you have to learn a whole new thing just to use that

**Comment by andob** [comment]
@sarbyn new version is on its way. I made a pull request with the changes, now new code is being polished and tested by @greenrobot-team . will be released soon I hope :)

**Comment by Malachiasz** [comment]
@greenrobot-team When will the separate android event bus artifact without support libraries be available? I see it's waiting for a review since July: https://github.com/greenrobot/EventBus/pull/670

**Comment by ooyellow58** [comment]
Have been patiently waited for the migration for long long time, hopefully it could be done soon. Thanks for all your effort!

**Comment by matejdro** [comment]
It looks like Google plans to get rid of Jetifier in the following AGP release: https://issuetracker.google.com/issues/141128458#comment13

**Comment by greenrobot-team** [comment]
A preview release that resolves this issue is now available: `3.3.0-SNAPSHOT`

If you want to help test this, you need to add the OSSRH snapshot repository in your `build.gradle.kts`:
```kotlin
allprojects {
    repositories {
        maven { url = uri("https://oss.sonatype.org/content/repositories/snapshots/") }
    }
}
```

**Comment by greenrobot-team** [comment]
This is now resolved as part of the `3.3.0` release.
